actions/rules/action_field_set.go

```diff
@@ -0,0 +1,91 @@
+package rules
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/fabric8-services/fabric8-wit/application"
+	"github.com/fabric8-services/fabric8-wit/convert"
+	"github.com/fabric8-services/fabric8-wit/workitem"
+)
+
+// ActionFieldSet takes a configuration JSON object that has field
+// names as the keys and a value as the argument. It updates the
+// given ChangeDetector and sets the Field[key] value to the
+// values given. Note that this only works on WorkItems.
+type ActionFieldSet struct {
+	Db     application.DB
+	Ctx    context.Context
+	UserID *uuid.UUID
+}
+
+// make sure the rule is implementing the interface.
+var _ Action = ActionFieldSet{}
+
+func (act ActionFieldSet) storeWorkItem(wi *workitem.WorkItem) (*workitem.WorkItem, error) {
+	if act.Ctx == nil {
+		return nil, errs.New("Context is nil")
+	}
+	if act.Db == nil {
+		return nil, errs.New("Database is nil")
+	}
+	if act.UserID == nil {
+		return nil, errs.New("UserID is nil")
+	}
+	var storeResultWorkItem *workitem.WorkItem
+	err := application.Transactional(act.Db, func(appl application.Application) error {
+		var err error
+		storeResultWorkItem, err = appl.WorkItems().Save(act.Ctx, wi.SpaceID, *wi, *act.UserID)
+		if err != nil {
+			return errs.Wrap(err, "Error updating work item")
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return storeResultWorkItem, nil
+}
+
+// OnChange executes the action rule.
+func (act ActionFieldSet) OnChange(newContext convert.ChangeDetector, contextChanges []convert.Change, configuration string, actionChanges *[]convert.Change) (convert.ChangeDetector, []convert.Change, error) {
+	// check if the newContext is a WorkItem, fail otherwise.
+	wiContext, ok := newContext.(workitem.WorkItem)
+	if !ok {
+		return nil, nil, errs.New("Given context is not a WorkItem: " + reflect.TypeOf(newContext).String())
+	}
+	// deserialize the config JSON
+	var rawType map[string]interface{}
+	err := json.Unmarshal([]byte(configuration), &rawType)
+	if err != nil {
+		return nil, nil, errs.Wrap(err, "Failed to unmarshall from action configuration to a map: "+configuration)
```

Comment:
all errors should by convention start with lower case letters.

Reply:
Sentences start with an uppercase letter in my opinion. Convention does not beat spelling rules. Sorry, I am a spelling nazi. It also massively bothers me that I can't end errors with a "." without golint complaining. This is just ridicoulus. Spelling rules are fundamental to communication.

But for now, I'll change that. Still, this is broken and always hits me in the eye when reading the code.

Reply:
> Sentences start with an uppercase letter in my opinion. Convention does not beat spelling rules. Sorry, I am a spelling nazi. 

The convention is there because you almost always don't know if your error is the last stage or if an error is wrapping that again. And then the wrapped error will print its own error message and the one you're writing here: `outer error: Failed to unmarshall from action...`. And sorry but the capital `F` after the colon is not following proper English rules.

See https://godoc.org/github.com/pkg/errors#hdr-Formatted_printing_of_errors:

> %s    print the error. If the error has a Cause it will be printed recursively

Code:
```go
cause := errors.New("whoops")
err := errors.Wrap(cause, "oh noes")
fmt.Println(err)
```
Outputs:
```
oh noes: whoops
```

Here you can read more about this convention and for why we have it: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

> It also massively bothers me that I can't end errors with a "." without golint complaining. This is just ridicoulus. Spelling rules are fundamental to communication.

Same comment as above.

> But for now, I'll change that. Still, this is broken and always hits me in the eye when reading the code.

Nothing is broken.

Reply:
Outdated. Already fixed.